relayer/relays/parachain/scanner.go

```diff
@@ -464,3 +479,159 @@ func (s *Scanner) findLatestNonce(ctx context.Context) (uint64, error) {
 	}
 	return ethInboundNonce, err
 }
+
+func (s *Scanner) IsBanned(m OutboundQueueMessage) (bool, error) {
+	destination, err := GetDestination(m)
+	if err != nil {
+		return true, err
+	}
+
+	return s.ofac.IsBanned(destination, "")
```

Comment:
Maybe add a comment that we will in a follow up ban the source address.

Reply:
>Does that require AliasOrigin with V2 first?

To extract it from the on-chain message, yes. In the interim we can fetch the event from Subscan and pull out the source from there.